p2p/module_raintree_test.go

```diff
@@ -1,202 +1,203 @@
 	//                                    val_10                                                                                                                   val_1                                                                                                       val_19
 	//            ┌──────────────────────────┴──────────────┬──────────────────────────────────────┐                                         ┌────────────────────────┴────────────┬──────────────────────────────────┐                                  ┌────────────────────────┴──────────────┬────────────────────────────────────┐
 	//          val_16                                    val_10                                 val_22                                     val_7                                 val_1                             val_13                             val_25                                  val_19                                val_4
 	//   ┌────────┴─────┬───────────┐              ┌────────┴─────┬───────────┐           ┌────────┴─────┬───────────┐              ┌────────┴────┬──────────┐             ┌───────┴────┬─────────┐          ┌────────┴─────┬──────────┐         ┌───────┴─────┬──────────┐             ┌────────┴─────┬───────────┐          ┌───────┴────┬──────────┐
 	// val_20         val_16      val_24         val_14         val_10      val_18      val_26         val_22      val_12         val_11        val_7      val_15        val_5        val_1     val_9      val_17         val_13     val_3     val_2         val_25     val_6         val_23         val_19      val_27     val_8        val_4      val_21
-	originatorNode := validatorId(t, 1)
-	var expectedCalls = TestRainTreeCommConfig{
-		validatorId(t, 1):  {0, 0}, // Originator
-		validatorId(t, 2):  {1, 1},
-		validatorId(t, 3):  {1, 1},
-		validatorId(t, 4):  {1, 1},
-		validatorId(t, 5):  {1, 1},
-		validatorId(t, 6):  {1, 1},
-		validatorId(t, 7):  {1, 1},
-		validatorId(t, 8):  {1, 1},
-		validatorId(t, 9):  {1, 1},
-		validatorId(t, 10): {1, 1},
-		validatorId(t, 11): {1, 1},
-		validatorId(t, 12): {1, 1},
-		validatorId(t, 13): {1, 1},
-		validatorId(t, 14): {1, 1},
-		validatorId(t, 15): {1, 1},
-		validatorId(t, 16): {1, 1},
-		validatorId(t, 17): {1, 1},
-		validatorId(t, 18): {1, 1},
-		validatorId(t, 19): {1, 1},
-		validatorId(t, 20): {1, 1},
-		validatorId(t, 21): {1, 1},
-		validatorId(t, 22): {1, 1},
-		validatorId(t, 23): {1, 1},
-		validatorId(t, 24): {1, 1},
-		validatorId(t, 25): {1, 1},
-		validatorId(t, 26): {1, 1},
-		validatorId(t, 27): {1, 1},
-	}
-	testRainTreeCalls(t, originatorNode, expectedCalls, false)
+	originatorNode := validatorId(1)
+	var expectedCalls = TestNetworkSimulationConfig{
+		originatorNode:  {0, 6},
```

Comment:
It's very strange to me that these numbers continue to change... This is the third time now. I think that this emphasizes further the need to move the `source of truth` code to this repository in order for formal verification. At this time, we are simply relying on these hard coded constants - which clearly are not stable

Reply:
> It's very strange to me that these numbers continue to change... This is the third time now. 

It is the second time they're changing. It's the same change as in #179 which was never finished & merged in.

I decided to reduce the scope of that PR into several smaller work tasks per the comment here: [the testing frame](https://github.com/pokt-network/pocket/pull/179#issuecomment-1281126451).

I believe my first approach was naive, but this second one makes more sense (with comments in some tests to help explain it):

```
func TestRainTreeNetworkCompleteFourNodes(t *testing.T) {
	// Test configurations (visualization retrieved from simulator)
	// 	                val_1
	// 	  ┌───────────────┴────┬─────────────────┐
	//  val_2                val_1             val_3
	//    └───────┐            └───────┐         └───────┐
	// 		    val_3                val_2             val_4
	originatorNode := validatorId(1)
	var expectedCalls = TestNetworkSimulationConfig{
		originatorNode: {0, 3}, // val_1 does 3 network writes (two to val_2 and 1 to val_3)
		validatorId(2): {2, 1}, // val_2 does 2 network reads (both from val_1) and 1 network write (to val_3)
		validatorId(3): {2, 1}, // val_2 does 2 network reads (from val_1 and val_2) and 1 network write (to val_4)
		validatorId(4): {1, 0}, // val_2 does 1 network read (from val_3)
	}
	testRainTreeCalls(t, originatorNode, expectedCalls)
}
```

Reply:
> It is the second time they're changing. It's the same change as in https://github.com/pokt-network/pocket/pull/179 which was never finished & merged in.

My mistake - sorry



Cool makes sense - can you link me the follow up issues that satisfy:
> I think that this emphasizes further the need to move the source of truth code to this repository in order for formal verification
I want to make sure this is well documented and the intention is clear before unblocking this review

Reply:
@jessicadaugherty Can you linkn to the ticket for (1) from https://github.com/pokt-network/pocket/pull/179#issuecomment-1281133270 when you have a chance?

Reply:
Yes will include as part of origin doc. Thanks @Olshansk!

Reply:
Sorry if I'm out of sync here: 

> Adding the Jupyter notebook to generate tests

Is this Ticket 1? I'm not sure if that satisfies my ask.

Waiting for this linking to approve

Reply:
I created #314 as follow up work specific to the tooling used to generate the tests.

Also added 2 TODOs in the code referencing it.